File: SampleCode/RestApi/Clients/Java/src/main/java/EmbyClient/Java/ItemRefreshServiceApi.java
/*
 * Emby Server REST API
 * 
 */

package EmbyClient.Java;

import EmbyClient.ApiCallback;
import EmbyClient.ApiClient;
import EmbyClient.ApiException;
import EmbyClient.ApiResponse;
import EmbyClient.Configuration;
import EmbyClient.Pair;
import EmbyClient.ProgressRequestBody;
import EmbyClient.ProgressResponseBody;

import com.google.gson.reflect.TypeToken;

import java.io.IOException;


import io.swagger.client.model.MetadataRefreshMode;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ItemRefreshServiceApi {
    private ApiClient apiClient;

    public ItemRefreshServiceApi() {
        this(Configuration.getDefaultApiClient());
    }

    public ItemRefreshServiceApi(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public ApiClient getApiClient() {
        return apiClient;
    }

    public void setApiClient(ApiClient apiClient) {
        this.apiClient = apiClient;
    }

    /**
     * Build call for postItemsByIdRefresh
     * @param id Item Id (required)
     * @param recursive Indicates if the refresh should occur recursively. (optional)
     * @param metadataRefreshMode Specifies the metadata refresh mode (optional)
     * @param imageRefreshMode Specifies the image refresh mode (optional)
     * @param replaceAllMetadata Determines if metadata should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param replaceAllImages Determines if images should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param progressListener Progress listener
     * @param progressRequestListener Progress request listener
     * @return Call to execute
     * @throws ApiException If fail to serialize the request body object
     */
    public com.squareup.okhttp.Call postItemsByIdRefreshCall(String id, Boolean recursive, MetadataRefreshMode metadataRefreshMode, MetadataRefreshMode imageRefreshMode, Boolean replaceAllMetadata, Boolean replaceAllImages, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        Object localVarPostBody = null;
        
        // create path and map variables
        String localVarPath = "/Items/{Id}/Refresh"
            .replaceAll("\\{" + "Id" + "\\}", apiClient.escapeString(id.toString()));

        List<Pair> localVarQueryParams = new ArrayList<Pair>();
        List<Pair> localVarCollectionQueryParams = new ArrayList<Pair>();
        if (recursive != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("Recursive", recursive));
        if (metadataRefreshMode != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("MetadataRefreshMode", metadataRefreshMode));
        if (imageRefreshMode != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("ImageRefreshMode", imageRefreshMode));
        if (replaceAllMetadata != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("ReplaceAllMetadata", replaceAllMetadata));
        if (replaceAllImages != null)
        localVarQueryParams.addAll(apiClient.parameterToPair("ReplaceAllImages", replaceAllImages));

        Map<String, String> localVarHeaderParams = new HashMap<String, String>();

        Map<String, Object> localVarFormParams = new HashMap<String, Object>();

        final String[] localVarAccepts = {
            
        };
        final String localVarAccept = apiClient.selectHeaderAccept(localVarAccepts);
        if (localVarAccept != null) localVarHeaderParams.put("Accept", localVarAccept);

        final String[] localVarContentTypes = {
            
        };
        final String localVarContentType = apiClient.selectHeaderContentType(localVarContentTypes);
        localVarHeaderParams.put("Content-Type", localVarContentType);

        if(progressListener != null) {
            apiClient.getHttpClient().networkInterceptors().add(new com.squareup.okhttp.Interceptor() {
                @Override
                public com.squareup.okhttp.Response intercept(com.squareup.okhttp.Interceptor.Chain chain) throws IOException {
                    com.squareup.okhttp.Response originalResponse = chain.proceed(chain.request());
                    return originalResponse.newBuilder()
                    .body(new ProgressResponseBody(originalResponse.body(), progressListener))
                    .build();
                }
            });
        }

        String[] localVarAuthNames = new String[] { "apikeyauth", "embyauth" };
        return apiClient.buildCall(localVarPath, "POST", localVarQueryParams, localVarCollectionQueryParams, localVarPostBody, localVarHeaderParams, localVarFormParams, localVarAuthNames, progressRequestListener);
    }
    
    @SuppressWarnings("rawtypes")
    private com.squareup.okhttp.Call postItemsByIdRefreshValidateBeforeCall(String id, Boolean recursive, MetadataRefreshMode metadataRefreshMode, MetadataRefreshMode imageRefreshMode, Boolean replaceAllMetadata, Boolean replaceAllImages, final ProgressResponseBody.ProgressListener progressListener, final ProgressRequestBody.ProgressRequestListener progressRequestListener) throws ApiException {
        // verify the required parameter 'id' is set
        if (id == null) {
            throw new ApiException("Missing the required parameter 'id' when calling postItemsByIdRefresh(Async)");
        }
        
        com.squareup.okhttp.Call call = postItemsByIdRefreshCall(id, recursive, metadataRefreshMode, imageRefreshMode, replaceAllMetadata, replaceAllImages, progressListener, progressRequestListener);
        return call;

        
        
        
        
    }

    /**
     * Refreshes metadata for an item
     * Requires authentication as user
     * @param id Item Id (required)
     * @param recursive Indicates if the refresh should occur recursively. (optional)
     * @param metadataRefreshMode Specifies the metadata refresh mode (optional)
     * @param imageRefreshMode Specifies the image refresh mode (optional)
     * @param replaceAllMetadata Determines if metadata should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param replaceAllImages Determines if images should be replaced. Only applicable if mode is FullRefresh (optional)
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public void postItemsByIdRefresh(String id, Boolean recursive, MetadataRefreshMode metadataRefreshMode, MetadataRefreshMode imageRefreshMode, Boolean replaceAllMetadata, Boolean replaceAllImages) throws ApiException {
        postItemsByIdRefreshWithHttpInfo(id, recursive, metadataRefreshMode, imageRefreshMode, replaceAllMetadata, replaceAllImages);
    }

    /**
     * Refreshes metadata for an item
     * Requires authentication as user
     * @param id Item Id (required)
     * @param recursive Indicates if the refresh should occur recursively. (optional)
     * @param metadataRefreshMode Specifies the metadata refresh mode (optional)
     * @param imageRefreshMode Specifies the image refresh mode (optional)
     * @param replaceAllMetadata Determines if metadata should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param replaceAllImages Determines if images should be replaced. Only applicable if mode is FullRefresh (optional)
     * @return ApiResponse&lt;Void&gt;
     * @throws ApiException If fail to call the API, e.g. server error or cannot deserialize the response body
     */
    public ApiResponse<Void> postItemsByIdRefreshWithHttpInfo(String id, Boolean recursive, MetadataRefreshMode metadataRefreshMode, MetadataRefreshMode imageRefreshMode, Boolean replaceAllMetadata, Boolean replaceAllImages) throws ApiException {
        com.squareup.okhttp.Call call = postItemsByIdRefreshValidateBeforeCall(id, recursive, metadataRefreshMode, imageRefreshMode, replaceAllMetadata, replaceAllImages, null, null);
        return apiClient.execute(call);
    }

    /**
     * Refreshes metadata for an item (asynchronously)
     * Requires authentication as user
     * @param id Item Id (required)
     * @param recursive Indicates if the refresh should occur recursively. (optional)
     * @param metadataRefreshMode Specifies the metadata refresh mode (optional)
     * @param imageRefreshMode Specifies the image refresh mode (optional)
     * @param replaceAllMetadata Determines if metadata should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param replaceAllImages Determines if images should be replaced. Only applicable if mode is FullRefresh (optional)
     * @param callback The callback to be executed when the API call finishes
     * @return The request call
     * @throws ApiException If fail to process the API call, e.g. serializing the request body object
     */
    public com.squareup.okhttp.Call postItemsByIdRefreshAsync(String id, Boolean recursive, MetadataRefreshMode metadataRefreshMode, MetadataRefreshMode imageRefreshMode, Boolean replaceAllMetadata, Boolean replaceAllImages, final ApiCallback<Void> callback) throws ApiException {

        ProgressResponseBody.ProgressListener progressListener = null;
        ProgressRequestBody.ProgressRequestListener progressRequestListener = null;

        if (callback != null) {
            progressListener = new ProgressResponseBody.ProgressListener() {
                @Override
                public void update(long bytesRead, long contentLength, boolean done) {
                    callback.onDownloadProgress(bytesRead, contentLength, done);
                }
            };

            progressRequestListener = new ProgressRequestBody.ProgressRequestListener() {
                @Override
                public void onRequestProgress(long bytesWritten, long contentLength, boolean done) {
                    callback.onUploadProgress(bytesWritten, contentLength, done);
                }
            };
        }

        com.squareup.okhttp.Call call = postItemsByIdRefreshValidateBeforeCall(id, recursive, metadataRefreshMode, imageRefreshMode, replaceAllMetadata, replaceAllImages, progressListener, progressRequestListener);
        apiClient.executeAsync(call, callback);
        return call;
    }
}

File: SampleCode/RestApi/Clients/Python/emby_client/api/item_refresh_service_api.py
# coding: utf-8

"""
    Emby Server REST API (BETA)
"""

from __future__ import absolute_import

import re  # noqa: F401

# python 2 and python 3 compatibility library
import six

from emby_client.api_client import ApiClient


class ItemRefreshServiceApi(object):
    """NOTE: This class is auto generated.
    """

    def __init__(self, api_client=None):
        if api_client is None:
            api_client = ApiClient()
        self.api_client = api_client

    def post_items_by_id_refresh(self, id, **kwargs):  # noqa: E501
        """Refreshes metadata for an item  # noqa: E501

        Requires authentication as user  # noqa: E501
        This method makes a synchronous HTTP request by default. To make an
        asynchronous HTTP request, please pass async_req=True
        >>> thread = api.post_items_by_id_refresh(id, async_req=True)
        >>> result = thread.get()

        :param async_req bool
        :param str id: Item Id (required)
        :param bool recursive: Indicates if the refresh should occur recursively.
        :param MetadataRefreshMode metadata_refresh_mode: Specifies the metadata refresh mode
        :param MetadataRefreshMode image_refresh_mode: Specifies the image refresh mode
        :param bool replace_all_metadata: Determines if metadata should be replaced. Only applicable if mode is FullRefresh
        :param bool replace_all_images: Determines if images should be replaced. Only applicable if mode is FullRefresh
        :return: None
                 If the method is called asynchronously,
                 returns the request thread.
        """
        kwargs['_return_http_data_only'] = True
        if kwargs.get('async_req'):
            return self.post_items_by_id_refresh_with_http_info(id, **kwargs)  # noqa: E501
        else:
            (data) = self.post_items_by_id_refresh_with_http_info(id, **kwargs)  # noqa: E501
            return data

    def post_items_by_id_refresh_with_http_info(self, id, **kwargs):  # noqa: E501
        """Refreshes metadata for an item  # noqa: E501

        Requires authentication as user  # noqa: E501
        This method makes a synchronous HTTP request by default. To make an
        asynchronous HTTP request, please pass async_req=True
        >>> thread = api.post_items_by_id_refresh_with_http_info(id, async_req=True)
        >>> result = thread.get()

        :param async_req bool
        :param str id: Item Id (required)
        :param bool recursive: Indicates if the refresh should occur recursively.
        :param MetadataRefreshMode metadata_refresh_mode: Specifies the metadata refresh mode
        :param MetadataRefreshMode image_refresh_mode: Specifies the image refresh mode
        :param bool replace_all_metadata: Determines if metadata should be replaced. Only applicable if mode is FullRefresh
        :param bool replace_all_images: Determines if images should be replaced. Only applicable if mode is FullRefresh
        :return: None
                 If the method is called asynchronously,
                 returns the request thread.
        """

        all_params = ['id', 'recursive', 'metadata_refresh_mode', 'image_refresh_mode', 'replace_all_metadata', 'replace_all_images']  # noqa: E501
        all_params.append('async_req')
        all_params.append('_return_http_data_only')
        all_params.append('_preload_content')
        all_params.append('_request_timeout')

        params = locals()
        for key, val in six.iteritems(params['kwargs']):
            if key not in all_params:
                raise TypeError(
                    "Got an unexpected keyword argument '%s'"
                    " to method post_items_by_id_refresh" % key
                )
            params[key] = val
        del params['kwargs']
        # verify the required parameter 'id' is set
        if ('id' not in params or
                params['id'] is None):
            raise ValueError("Missing the required parameter `id` when calling `post_items_by_id_refresh`")  # noqa: E501

        collection_formats = {}

        path_params = {}
        if 'id' in params:
            path_params['Id'] = params['id']  # noqa: E501

        query_params = []
        if 'recursive' in params:
            query_params.append(('Recursive', params['recursive']))  # noqa: E501
        if 'metadata_refresh_mode' in params:
            query_params.append(('MetadataRefreshMode', params['metadata_refresh_mode']))  # noqa: E501
        if 'image_refresh_mode' in params:
            query_params.append(('ImageRefreshMode', params['image_refresh_mode']))  # noqa: E501
        if 'replace_all_metadata' in params:
            query_params.append(('ReplaceAllMetadata', params['replace_all_metadata']))  # noqa: E501
        if 'replace_all_images' in params:
            query_params.append(('ReplaceAllImages', params['replace_all_images']))  # noqa: E501

        header_params = {}

        form_params = []
        local_var_files = {}

        body_params = None
        # Authentication setting
        auth_settings = ['apikeyauth', 'embyauth']  # noqa: E501

        return self.api_client.call_api(
            '/Items/{Id}/Refresh', 'POST',
            path_params,
            query_params,
            header_params,
            body=body_params,
            post_params=form_params,
            files=local_var_files,
            response_type=None,  # noqa: E501
            auth_settings=auth_settings,
            async_req=params.get('async_req'),
            _return_http_data_only=params.get('_return_http_data_only'),
            _preload_content=params.get('_preload_content', True),
            _request_timeout=params.get('_request_timeout'),
            collection_formats=collection_formats)
